### PR TITLE
Avoid using the global window

### PR DESCRIPTION
**Pull request by FFKL**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/ngneat/dirty-check-forms/issues/13

## What is the new behavior?

`window ` global object is included in config.

## Does this PR introduce a breaking change?

```
[x] Yes
[ ] No

### CODE REVIEW

**Comment by FFKL** [comment]
@NetanelBasal How can I implement test cases? As I understand, I can't check native window behavior in cypress. What do you think about it?

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ Call the `dirtyCheck` function, which accepts three arguments:
 3. Config:  
   - `debounce` - debounce time of `valueChanges`. Defaults to 300  
   - `withDisabled` - whether to include disable fields (by using control's `getRawValue`) or not. Defaults to `true`. 
+  - `window` - global `Window` object to handle `onbeforeunload` event. Defaults to `undefined`.
```
Comment (NetanelBasal) [inline README.md:47]: This is a breaking change

File: libs/dirty-check-forms/src/lib/dirty-check.ts [hunk 2]
```diff
@@ -59,16 +60,18 @@ export function dirtyCheck<U>(
...
-      fromEvent(window, 'beforeunload')
-        .pipe(withLatestFrom(isDirty$))
-        .subscribe(([event, isDirty]) => {
-          if (isDirty) {
-            event.preventDefault();
-            event.returnValue = false;
-          }
-        })
-    );
+    if (window) {
```
Comment (NetanelBasal) [inline libs/dirty-check-forms/src/lib/dirty-check.ts:63]: Why not using isPlatformBrowser, and that's it?

File: README.md (hunk 1 above)
Comment (FFKL) [reply README.md:47]: Yes, I know. Is it impossible to mark current behavior as deprecated?

File: libs/dirty-check-forms/src/lib/dirty-check.ts (hunk 2 above)
Comment (FFKL) [reply libs/dirty-check-forms/src/lib/dirty-check.ts:63]: I think it's better not to use globals and pass all module dependencies as parameters. Also, dirty check doesn't support checking without using the `beforeunload` event. Maybe `useBeforeunloadEvent` parameter can be used as a compromise?

Comment (NetanelBasal) [reply libs/dirty-check-forms/src/lib/dirty-check.ts:63]: Yes, we can add useBeforeunloadEvent, but use isPlatformBrowser to protect from SSR

**Comment by FFKL** [comment]
@NetanelBasal I've added `useBeforeunloadEvent`. Is it correct?

**Pull request merged**